WooCommerce/UITestsFoundation/Screens/Products/SingleProductScreen.swift

```diff
@@ -41,10 +41,25 @@ public final class SingleProductScreen: ScreenObject {
         return self
     }
 
-    public func verifyNewProductScreenLoaded(productName: String) {
+    public func verifyPublishedProductScreenLoaded(productType: String, productName: String) {
+        let priceLabel = NSPredicate(format: "label == 'Price'")
+        let addVariationLabel = NSPredicate(format: "label == 'Add variations'")
+
+        // common fields on a published product screen
         XCTAssertTrue(app.buttons["save-product-button"].exists)
         XCTAssertTrue(app.staticTexts["TIP"].exists)
         XCTAssertTrue(app.textViews[productName].exists)
+
+        // different product types display different fields on the published product screen
+        // this is to validate that the correct screens are displayed
+        switch productType {
+        case "physical", "virtual":
+            XCTAssertTrue(app.staticTexts.containing(priceLabel).firstMatch.exists)
+        case "variable":
+            XCTAssertTrue(app.staticTexts.containing(addVariationLabel).firstMatch.exists)
```

Comment:
This will work too:

```suggestion
        case "physical", "virtual":
            XCTAssertTrue(app.staticTexts["Price"].exists)
        case "variable":
            XCTAssertTrue(app.staticTexts["Add variations"].exists)
```

Besides being shorter and allowing to remove two `NSPredicate`s at the function beginning, it's also stricter - if there will be more than one match, assertion will fail. With `firstMatch`, the assertion will not care it's there's more than one match.

Reply:
thanks for this suggestion! yeah makes sense and it reads better too. updated this and a similar function to use this way for exact match labels in https://github.com/woocommerce/woocommerce-ios/pull/9441/commits/8404f5679e27a4237a0e75a955b7bd6bfbfe7262